src/realm/spec.cpp

```diff
@@ -31,6 +31,13 @@ Spec::~Spec() noexcept
     }
 }
 
+void Spec::init(ref_type ref) noexcept
+{
+    if (!m_top.is_attached() || m_top.get_ref() != ref) {
```

Comment:
The usual comment for this type of expression; I would like to see this extracted to a variable with a meaningful name, such that it becomes trivially readable.